lib/msprime.c

```diff
@@ -4736,6 +4736,14 @@ msp_run_sweep(msp_t *self)
         goto out;
     }
 
+    // let curr_step = 1 before the loops and then add 1 to it at the end of
```

Comment:
This seems correct to me @gbinux, and this is a good explanation. However, I think the comment will be confusing to people who read the code in the future, who have no idea what the old way was. Comments should describe the code *as it is*.

Reply:
Maybe we don't need the comment at all?